### DISCUSSION TITLE
A 'AnisoHeatConductionMaterial' is not a registered object.

### DISCUSSION THREAD

**Discussion by hamedattariani** [idaholab/moose#26035] [Q&A Modules: General] [answered]
Moose was compiled successfully and passed all tests. 
I only replaced the AnisoHeatConductionMaterial.C and .h files with the following versions.
https://github.com/idaholab/moose/pull/22028/files

I created an application, changed the heat transfer module to "yes" in the makefile, and compiled it. It gave me the executable files.
Now, if I use the "AnisoHeatConductionMaterial" in the input script, it gives me the following error.
```
*** ERROR ***
A 'AnisoHeatConductionMaterialM' is not a registered object.

If you are trying to find this object in a dynamically loaded library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
```
I checked the metrials_Unity.C file in moose/modules/heat_transfer/build folder and the material was listed in the file:
```
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/AnisoHeatConductionMaterial.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/ElectricalConductivity.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/FunctionPathEllipsoidHeatSource.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/GapConductance.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/GapConductanceConstant.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/HeatConductionMaterial.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/SemiconductorLinearConductivity.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/SideSetHeatTransferMaterial.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/ThermalCompliance.C"
#include"/home/hamed/projects/moose/modules/heat_transfer/src/materials/ThermalSensitivity.C"
```
Any suggestion?

**Comment by lindsayad** [comment]
When you modified it, did you modify it in `heat_conduction` or `heat_transfer`? That PR is quite old

**Reply by hamedattariani** [reply]
I modified it on heat_transfer. Heat Conduction module contains short-cuts to heat_transfer folder. 
If I click on folders on heat_conduction module, then it navigates me to heat transfer module folder which I updated the file.

**Reply by lindsayad** [reply]
What is the output of `git status` in your moose directory?

**Reply by hamedattariani** [reply]
efresh index: 100% (38514/38514), done.
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   framework/include/auxkernels/MaterialRealTensorValueAux.h
	modified:   framework/src/auxkernels/MaterialRealTensorValueAux.C
	modified:   modules/heat_transfer/include/materials/AnisoHeatConductionMaterial.h
	modified:   modules/heat_transfer/src/materials/AnisoHeatConductionMaterial.C

no changes added to commit (use "git add" and/or "git commit -a")

**Reply by lindsayad** [reply]
If you're on Mac, please paste the output of `otool -L your_executable | grep heat` (replacing `your_executable` with the name of your executable), and if on Linux the output of `ldd your_executable | grep heat`

**Reply by hamedattariani** [reply]
libheat_transfer-opt.so.0 => /home/hamed/projects/moose/modules/heat_transfer/lib/libheat_transfer-opt.so.0 (0x00007ff4dae00000)

**Reply by lindsayad** [reply]
This all looks fine to me. My only remaining suggestion is to try running `make clobberall` in your application and then building again.

**Comment by GiudGiud** [comment]
'AnisoHeatConductionMaterialM' is there a typo with an extra M at the end?

**Reply by hamedattariani** [reply, marked as answer]
Thanks for checking that. That was not the issue. I tried multiple ways to fix the registration. One method was to define it as a custom-made kernel with a different name (AnisoHeatConductionMaterialM), but that did not work, either. The error stems from that try.

I resolved it by simply changing the   "registerMooseObject("HeatConductionApp", AnisoHeatConductionMaterial);" to "registerMooseObject("HeatTranferApp", AnisoHeatConductionMaterial);" in the C file.

**Reply by GiudGiud** [reply]
ah yes that is definitely the correct fix. Thank you for reporting back
